.eslintrc.js

```diff
@@ -38,48 +38,100 @@ module.exports = {
     'plugin:react-hooks/recommended',
   ],
 
-  plugins: [
-    '@babel',
-    'react',
-    'import',
-  ],
+  plugins: ['@babel', 'react', 'import', 'prettier'],
 
   globals: {
     document: 'readonly',
     window: 'readonly',
   },
 
   rules: {
+    // Prettier changes and reasoning
+
+    'prettier/prettier': 'error',
+
+    // Our usage of spaces before *named* function parens is unusual, and
+    // doesn't match the prettier spec. prettier does not offer an option
+    // to configure this
+    'space-before-function-paren': [
+      'error',
+      { anonymous: 'always', named: 'never' },
+    ],
+    // 'lines-around-comment': 'error',
+    // Our eslint config has the default setting for this as error. This
+    // include beforeBlockComment: true, but in order to match the prettier
+    // spec you have to enable before and after blocks, objects and arrays
+    // https://github.com/prettier/eslint-config-prettier#lines-around-comment
+    'lines-around-comment': [
+      'error',
+      {
+        beforeBlockComment: true,
+        afterLineComment: false,
+        allowBlockStart: true,
+        allowBlockEnd: true,
+        allowObjectStart: true,
+        allowObjectEnd: true,
+        allowArrayStart: true,
+        allowArrayEnd: true,
+      },
+    ],
+    // Prettier has some opinions on mixed-operators, and there is ongoing work
+    // to make the output code clear. It is better today then it was when the first
+    // PR to add prettier. That being said, the workaround for keeping this rule enabled
+    // requires breaking parts of operations into different variables -- which I believe
+    // to be worse. https://github.com/prettier/eslint-config-prettier#no-mixed-operators
+    'no-mixed-operators': 'off',
+    // Prettier wraps single line functions with ternaries, etc in parens by default, but
+    // if the line is long enough it breaks it into a separate line and removes the parens.
+    // The second behavior conflicts with prettier. There is some guides on the repo about
+    // how you can keep it enabled:
+    // https://github.com/prettier/eslint-config-prettier#no-confusing-arrow
+    // However, in practice this conflicts with prettier adding parens around short lines,
+    // when autofixing in vscode and others.
+    'no-confusing-arrow': 'off',
+    // There is no configuration in prettier for how it stylizes regexes, which conflicts
+    // with wrap-regex.
+    'wrap-regex': 'off',
+    // Prettier handles all indentation automagically. it can be configured here
+    // https://prettier.io/docs/en/options.html#tab-width but the default matches our
+    // style.
+    indent: 'off',
+    'function-paren-newline': 'off',
```

Comment:
Was there an explanation somewhere for why these three rules were disabled? :thinking: